### PR TITLE
hostcall: get_property

### PR DESCRIPTION
**Pull request by eguzki**

### CODE REVIEW

**Comment by eguzki** [comment]
ready for review @alexsnaps

File: src/expectations.rs [hunk 1]
```diff
@@ -649,4 +653,36 @@ impl Expect {
...
     }
+
+    pub fn set_expect_get_property(
+        &mut self,
+        path: Option<Vec<&str>>,
+        property_data: Option<&[u8]>,
+    ) {
+        self.expect_count += 1;
+        self.get_property_value.push((
+            path.map(|segments: Vec<&str>| serialize_property_path(segments)),
```
Comment (alexsnaps) [inline src/expectations.rs:664]: I wonder if I'd not keep the path as a Vec until the comparison is made, possibly making it easier to debug when the unexpected path gets queried?

**Review by alexsnaps — APPROVED** [review]
Only had a chance to have a real quick look before you're now leaving on vacation. So, other than one comment, lgtm! ;)
Also, would you mind `--signoff` the commit and use you RH email if that's not the case?

File: src/expectations.rs (hunk 1 above)
Comment (eguzki) [reply src/expectations.rs:664]: I came up with this changes to implement your proposition. But I do not see it making easier to debug. The interesting path to be debugged is the one coming from the VM, that that value always stays serialized.

```diff
diff --git a/src/expectations.rs b/src/expectations.rs
index 53a463b..513dd75 100644
--- a/src/expectations.rs
+++ b/src/expectations.rs
@@ -98,7 +98,7 @@ pub struct Expect {
         Option<Duration>,
         Option<u32>,
     )>,
-    get_property_value: Vec<(Option<Bytes>, Option<Bytes>)>,
+    get_property_value: Vec<(Option<Vec<String>>, Option<Bytes>)>,
 }
 
 impl Expect {
@@ -661,7 +661,7 @@ impl Expect {
     ) {
         self.expect_count += 1;
         self.get_property_value.push((
-            path.map(|segments: Vec<&str>| serialize_property_path(segments)),
+            path.map(|segments: Vec<&str>| segments.iter().map(|s| s.to_string()).collect()),
             property_data.map(|data| data.to_vec()),
         ));
     }
@@ -679,7 +679,11 @@ impl Expect {
                 self.expect_count -= 1;
                 let (expected_path, result) = self.get_property_value.remove(0);
                 // when expected anything is None, behave as catch all
-                let expected = expected_path.map(|p| p == path).unwrap_or(true);
+                let expected = expected_path
+                    .map(|p| {
+                        serialize_property_path(p.iter().map(|x| x.as_ref()).collect()) == path
+                    })
+                    .unwrap_or(true);
                 set_expect_status(expected);
                 result
             }
```

**Pull request merged**